### ISSUE TITLE
ApiKey and octopusServerUrl Mandatory

### ISSUE THREAD

**Issue by kottimon** [OctopusDeploy/OctopusDSC#89]
Hi!

In cTentacleAgent.psm1 and the function New-Tentacle the parameters $apiKey and $octopusServerUrl are mandatory and it seems to me they should not be?
I assume that those parameters are only really used if $RegisterWithServer = $true ?

And if I set $RegisterWithServer = $false and omit $apikey and $octopusServerUrl from cTentacleAgent properties, the LCM will exit with the error:
VERBOSE: [$Name]:                            [[cTentacleAgent]OctopusTentacle] Installing Tentacle...
Cannot bind argument to parameter 'apiKey' because it is an empty string.
    + CategoryInfo          : InvalidData: (:) [], CimException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,New-Tentacle
    + PSComputerName        : localhost

However, if I change the param-block to:
```
[Parameter(Mandatory = $True)]
        [string]$name,
        [Parameter(Mandatory = $False)]
        [string]$apiKey,
        [Parameter(Mandatory = $False)]
        [string]$octopusServerUrl
```
The configuration is applied successfully.

**Comment by matt-richardson** [comment]
Another good catch. 
Fixed in `3.0.78`.

**Comment by kottimon** [comment]
Very nice, thank you!
